crates/libcgroups/src/systemd/dbus_native/dbus.rs

```diff
@@ -296,20 +303,47 @@ impl DbusConnection {
             None,
         )?;
 
-        let reply = self.receive_complete_response()?;
-
-        // note that a single received response can contain multiple
-        // messages, so we must deserialize it piece by piece
         let mut ret = Vec::new();
-        let mut buf = &reply[..];
-
-        while !buf.is_empty() {
-            let mut ctr = 0;
-            let msg = Message::deserialize(&buf[ctr..], &mut ctr)?;
-            // we reset the buf, because I couldn't figure out how the adjust_counter function
-            // should should be changed to work correctly with non-zero start counter, and this solved that issue
-            buf = &buf[ctr..];
-            ret.push(msg);
+
+        // it is possible that while receiving messages, we get some extra/previous message
```

Comment:
Do we know why we get these extra messages?

Reply:
As of now the only extra messages that are observed are of type signal, which basically broadcast some event to all connected clients. The thing is we used to get these before as well, but usually these would get consumed in buffers of some other method calls or such, and get ignored. In few cases that would not happen, which lead to the dbus error in question. To solve this, now we keep consuming all messages until we get method return or error message (in case of method call message).